designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/ScenarioStatistics.scala

```diff
@@ -0,0 +1,138 @@
+package pl.touk.nussknacker.ui.statistics
+
+import cats.implicits.{toFoldableOps, toTraverseOps}
+import pl.touk.nussknacker.engine.api.component.ComponentType
+import pl.touk.nussknacker.restmodel.component
+import pl.touk.nussknacker.ui.process.repository.DbProcessActivityRepository
+
+import java.time.Instant
+
+object ScenarioStatistics {
+
+  def getGeneralStatistics(scenariosInputData: List[ScenarioStatisticsInputData]): Map[String, String] = {
+    if (scenariosInputData.isEmpty) {
+      Map.empty
+    } else {
+      val scenariosCount = scenariosInputData.count(!_.isFragment)
+      //        Nodes stats
+      val sortedNodes  = scenariosInputData.map(_.nodesCount).sorted
+      val nodesMedian  = sortedNodes.get(sortedNodes.length / 2).getOrElse(0)
+      val nodesAverage = sortedNodes.sum / scenariosInputData.length
+      val nodesMax     = sortedNodes.head
+      val nodesMin     = sortedNodes.last
+      //        Category stats
+      val categoriesCount = scenariosInputData.map(_.scenarioCategory).toSet.size
+      //        Version stats
+      val sortedVersions  = scenariosInputData.map(_.scenarioVersion.value).sorted
+      val versionsMedian  = sortedVersions.get(sortedVersions.length / 2).getOrElse(0)
+      val versionsAverage = sortedVersions.sum / scenariosInputData.length
+      val versionsMax     = sortedVersions.head
+      val versionsMin     = sortedVersions.last
+      //        Author stats
+      val authorsCount = scenariosInputData.map(_.createdBy).toSet.size
+      //        Fragment stats
+      val fragments        = scenariosInputData.map(_.fragmentsUsedCount).sorted
+      val fragmentsMedian  = fragments.get(fragments.length / 2).getOrElse(0)
+      val fragmentsAverage = fragments.sum / scenariosCount
+      //          Uptime stats
+      val lastActions = scenariosInputData.map(_.lastDeployedAction).filter(_.isDefined).sequence.get
```

Comment:
IMHO is not the correct way to calculate an uptime. `lastDeployedAction` is a deploy action but is necessarily the last one (you can check it here:pl.touk.nussknacker.ui.process.repository.DBFetchingProcessRepository.fetchLatestProcessDetailsByQueryAction), fe. the last one can be cancel. IMHO we should take all actions and if the last one is the deploy one then we can calculate the uptime, otherwise we cannot do any calculations.